R/table-transpose.R

```diff
@@ -91,6 +94,23 @@ transposeMappedDimnames <- function(mapped.dimnames) {
     mapped.dimnames
 }
 
+transposeCellText <- function(cell.text) {
+    if (!is.array(cell.text)) {
+        return(NULL)
+    }
+
+    d <- dim(cell.text)
+    n.dim <- length(d)
+    if (n.dim == 2 && d[1] == 1) {
+        cell.text <- array(cell.text)
```

Comment:
If `cell.text` is 2d (1xn) then the transpose should be (nx1)? This code makes is a 1d of length n array?

Reply:
Yes, I thought it would be simpler if it was always an array, rather than a vector sometimes. The C# code assumes this (so I can get the dim)

Reply:
In this case it already is an array though and an 1xn becomes n when transposed instead of nx1?

Reply:
```
 a <- array(1:3, dim = c(1, 3))
a
     [,1] [,2] [,3]
[1,]    1    2    3
array(a)
[1] 1 2 3
array(a) |> str()
 int [1:3(1d)] 1 2 3
```

Reply:
The consequence being
```
t(array(a))
     [,1] [,2] [,3]
[1,]    1    2    3
```
which is the same as the original `a` (it doesn't get transposed)?

Reply:
Actually it is because `t(cell.text)` becomes a 2D array in this case, whereas I want a 1D array. I'll add a comment.